monad-logger/Control/Monad/Logger.hs

```diff
@@ -579,6 +582,22 @@ runStderrLoggingT = (`runLoggingT` defaultOutput stderr)
 runStdoutLoggingT :: MonadIO m => LoggingT m a -> m a
 runStdoutLoggingT = (`runLoggingT` defaultOutput stdout)
 
+-- | Run a block using a @MonadLogger@ instance which writes tuples to an unbounded channel.
+--
+-- Since VERSION
+runChanLoggingT :: MonadIO m => Chan (Loc, LogSource, LogLevel, LogStr) -> LoggingT m a -> m a
+runChanLoggingT chan = (`runLoggingT` sink chan)
+    where
+        sink chan loc src lvl msg = writeChan chan (loc,src,lvl,msg)
+
+-- | Read tuples from an unbounded channel and log them, forever.
+--
+-- Since VERSION
+unChanLoggingT :: MonadLoggerIO m => Chan (Loc, LogSource, LogLevel, LogStr) -> m ()
```

Comment:
Then it should probably be `(MonadLogger m, MonadIO m)`. `MonadLoggerIO` itself seems unnecessary.